ondevice_learning.qmd

```diff
@@ -1,61 +1,630 @@
+
+To mitigate these risks, several defenses can be employed:
+
+* **Data Validation and Sanitization:** Before incorporating new data into the training dataset, it should be thoroughly validated and sanitized to ensure it is not malicious.
+* **Adversarial Training:** The model can be trained on adversarial examples to make it more robust to these types of attacks.
+* **Input Validation:** During inference, inputs should be validated to ensure they have not been manipulated to create adversarial examples.
+* **Regular Auditing and Monitoring:** Regularly auditing and monitoring the model's behavior can help to detect and mitigate adversarial attacks. In the context of tiny ML systems, this is easier said than done, because it is often hard to monitor embedded machine learning systems at the endpoint due to communication bandwidth limitations and so forth, which we will discuss in the MLOps chapter.
+
+By understanding the potential risks and implementing these defenses, we can help to secure on-device training at the endpoint/edge and mitigate the impact of adversarial attacks. Most people easily confuse data poisoning and adversarial attacks. So here is a table comparing data poisoning and adversarial attacks:
+
+| Aspect             | Data Poisoning                          | Adversarial Attacks                      |
+|--------------------|-----------------------------------------|-------------------------------------------|
+| **Timing**         | Training phase                          | Inference phase                          |
+| **Target**         | Training data                           | Input data                               |
+| **Goal**           | Negatively affect model's performance   | Cause incorrect predictions              |
+| **Method**         | Insert malicious examples into training data, often with incorrect labels | Add carefully crafted noise to input data |
+| **Example**        | Adding images of cats labeled as dogs to a dataset used for training an image classification model | Adding a small amount of noise to an image in a way that causes a face recognition system to misidentify a person |
+| **Potential Effects** | Model learns incorrect patterns and makes incorrect predictions | Immediate and potentially dangerous incorrect predictions |
+| **Applications Affected** | Any machine learning model | Autonomous vehicles, security systems, etc. |
+
+
+### On-Device Learning Security Concerns
```

Comment:
Could include topics or mentions of hardware security as well (side channel attacks)
https://dl.acm.org/doi/pdf/10.1145/3394885.3431639